quest_test/test_serializer.py

```diff
@@ -0,0 +1,69 @@
+import asyncio
+from dataclasses import dataclass
+import pytest
+from typing import Tuple, Dict, Any, Callable, TypeVar
+
+from quest.historian import Historian
+from quest.serializer import MasterSerializer, StepSerializer
+from quest.wrappers import step
+
+T = TypeVar('T')
+
+# Define custom class
+@dataclass
+class Stuff:
+    name: str
+    number: int
+
+    def make_statement(self):
+        return (self.name + ' ') * self.number
+
+
+# TypeSerializer for Stuff
+class StuffSerializer:
+    async def serialize(self, obj: Stuff) -> Tuple[Callable[..., T], Tuple[Any, ...], Dict[str, Any]]:
+        factory = Stuff
+        args = (obj.name, obj.number)
+        kwargs = {}
+        return factory, args, kwargs
```

Comment:
Because `deserialize` handles recreating the object, we no longer need to return `factory`

```py
async def serialize(self, obj):
    return (obj.name, obj.number), {}
```